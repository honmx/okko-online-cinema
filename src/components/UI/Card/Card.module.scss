@import "../../../styles/variables.scss";

.cardWrapper {
  position: relative;
  transition: scale 0.15s ease;
  height: 100%;

  &:hover {
    scale: 1.05;
    z-index: 10000000;
  }
}

.cardWithTrailer {
  margin-bottom: -40px;
  background-color: $primaryDarkGreyColor;
  border-radius: 25px;
  // height: 100%;

  & .card {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
  }
}

.card {
  // height: 100%;
  width: min(25vw, 350px);
  overflow: hidden;
  border-radius: 25px;

  video {
    object-fit: cover;
  }

}

@media screen and (max-width: 960px) {
  .card, .movieInfo {
    width: min(65vw, 300px);
  }
}

@media screen and (max-width: 375px) {
  .card, .movieInfo {
    width: min(85vw, 290px);
  }
}

.img {
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.movieTitle {
  color: $accentWhiteFontColor;
}

.sound {
  position: absolute;
  bottom: 2px;
  right: 2px;

  &:hover {
    &>div {
      &>div {
        background-color: $primaryDarkGreyColor !important;
      }
    }
  }

  img {
    width: 14px;
    height: 14px;
  }
}

.movieInfo {
  position: absolute;
  top: 100%;
  left: 0;
  right: 0;
  padding: 5px 20px;
  background-color: $primaryDarkGreyColor;
  border-bottom-left-radius: 25px;
  border-bottom-right-radius: 25px;
}

.upperLine {
  display: flex;
  align-items: center;
}

.lowerLine {
  display: flex;
  align-items: center;
  margin-top: 3px;
  color: $primaryFontColor;
  font-size: 12px;

  &>*:not(:last-child) {
    margin-right: 10px;
  }
}

.ratingContainer {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  margin-right: 10px;

  .rating {
    padding: 2px 4px;
    color: $accentWhiteFontColor;
    border-radius: 2px;
    font-size: 9px;
  }

  .usualRate {
    background-color: $primaryGreyColor;
  }

  .greenRate {
    background-color: $greenColor;
  }
}