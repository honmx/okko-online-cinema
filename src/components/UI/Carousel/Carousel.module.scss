@import "../../../styles/variables.scss";

.carouselContainer {
  
}

.link {
  display: inline-flex;
  align-items: center;
  color: inherit;
  text-decoration: none;

  &:hover {
    .arrow {
      opacity: 1;
      display: inline-block;
      margin-left: 10px;
    }
  }
}

.arrow {
  opacity: 0;
  margin-left: 5px;
  margin-top: 2px;
  rotate: 90deg;
  width: 10px;
  height: 10px;
  transition: all 0.15s ease;
}

.listContainer {
  position: relative;

  &:hover {
    .carouselButton {
      display: flex;
    }
  }
}

.carouselButton {
  display: none;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  padding: 30px 15px;
  background-color: rgba($color: $primaryGreyColor, $alpha: 0.5);
  border-radius: 12px;
  border: none;
  transition: 0.15s ease all;
  backdrop-filter: blur(5px);
  cursor: pointer;
  z-index: 1000;

  &:hover {
    scale: 1.1;
  }
}

.leftButton {
  left: 10px;

  img {
    rotate: -90deg;
  }
}

.rightButton {
  right: 10px;
  img {
    rotate: 90deg;
  }
}

.childrenContainer {
  display: flex;
  overflow: auto;
  margin-inline: -15px;
  margin-top: -20px;
  padding: 55px 15px;

  &::-webkit-scrollbar {
    display: none;
  }

  &>*:not(:last-child) {
    margin-right: 15px;
  }
}

.titleContainer {
  display: flex;
  align-items: center;
}

.image {
  max-height: 30px;
  width: auto;
  margin-right: 5px;
}