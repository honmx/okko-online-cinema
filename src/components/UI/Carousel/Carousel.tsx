import React, { FC, ReactNode } from "react";
import Title from "../Title/Title";
import Link from "next/link";
import Image from "next/image";
import arrow from "../../../assets/arrow.svg";
import s from "./Carousel.module.scss";
import { useCarouselScroll } from "../../../hooks/useCarouselScroll";

interface Props {
  title?: string;
  linkHref?: string;
  image?: string;
  className?: string;
  onClick?: () => void;
  children: ReactNode;
}

const Carousel: FC<Props> = ({ title, linkHref, image, className, children, onClick }) => {

  const { ref, isAbleToScrollLeft, isAbleToScrollRight } = useCarouselScroll();

  const handleLeftButtonClick = () => {
    if (!ref.current) return;

    ref.current.scrollTo({
      left: ref.current.scrollLeft - ref.current.clientWidth,
      top: 0,
      behavior: "smooth",
    });
  }

  const handleRightButtonClick = () => {
    if (!ref.current) return;

    ref.current.scrollTo({
      left: ref.current.scrollLeft + ref.current.clientWidth,
      top: 0,
      behavior: "smooth",
    });
  }

  return (
    <div className={`${s.carouselContainer} ${className}`}>
      <div className={s.titleContainer} onClick={onClick}>
        {
          title && !linkHref && image &&
          <>
            <Image src={image} alt="image" className={s.image} />
            <Title variant="h2" className={s.title}>{title}</Title>
          </>
        }
        {
          title && linkHref && !image &&
          <Link href={linkHref} className={s.link}>
            <Title variant="h2" className={s.title}>{title}</Title>
            <Image src={arrow} alt="arrow" className={s.arrow} />
          </Link>
        }
        {
          title && !linkHref && !image &&
          <Title variant="h2" className={s.title}>{title}</Title>
        }
      </div>
      <div className={s.listContainer}>
        {
          isAbleToScrollLeft &&
          <div className={s.carouselButtonWrapper}>
            <button className={`${s.carouselButton} ${s.leftButton}`} onClick={handleLeftButtonClick}>
              <Image src={arrow} alt="arrow" />
            </button>
          </div>
        }
        <div className={s.childrenContainer} ref={ref}>
          {children}
        </div>
        {
          isAbleToScrollRight &&
          <div className={s.carouselButtonWrapper}>
            <button className={`${s.carouselButton} ${s.rightButton}`} onClick={handleRightButtonClick}>
              <Image src={arrow} alt="arrow" />
            </button>
          </div>
        }
      </div>
    </div>
  )
};

export default Carousel;
