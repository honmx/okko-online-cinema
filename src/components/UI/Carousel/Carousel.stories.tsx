import { Meta, StoryObj } from "@storybook/react";
import CarouselComponent from "./Carousel";
import Card from "../Card/Card";
import { movies } from "../../../helpers/data/movies";

const meta: Meta<typeof CarouselComponent> = {
  title: "Carousel",
  component: CarouselComponent,
  tags: ["autodocs"],
  argTypes: {}
}

export default meta;

type Story = StoryObj<typeof CarouselComponent>

export const CarouselOfCardsWithoutTitle: Story = {
  args: {
    children: movies.map(movie => <Card key={movie.id} item={movie} linkHref="/" />)
  }
}

export const CarouselOfCardsWithTitle: Story = {
  args: {
    title: "Фильмы",
    linkHref: "/",
    children: movies.map(movie => <Card key={movie.id} item={movie} linkHref="/" />)
  }
}