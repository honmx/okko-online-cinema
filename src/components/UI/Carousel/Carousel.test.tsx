import { fireEvent, render } from "@testing-library/react";
import Carousel from "./Carousel";
import { movies } from "@/helpers/data/movies";
import Card from "../Card/Card";

describe("", () => {
  test("component renders", () => {
    const { container } = render(
      <Carousel>
        {movies.map(movie => <Card key={movie.id} item={movie} linkHref={movie.title} />)}
      </Carousel>
    );
    
    expect(container.getElementsByClassName("carouselContainer")[0]).toBeInTheDocument();
  });
  
  test("carousel matches snapshot", () => {
    const component = render(
      <Carousel>
        {movies.map(movie => <Card key={movie.id} item={movie} linkHref={movie.title} />)}
      </Carousel>
    );
    
    expect(component).toMatchSnapshot();
  })
})