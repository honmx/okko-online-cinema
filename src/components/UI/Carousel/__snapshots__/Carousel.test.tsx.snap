// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` carousel matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carouselContainer undefined"
      >
        <div
          class="titleContainer"
        />
        <div
          class="listContainer"
        >
          <div
            class="childrenContainer"
          >
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="1+1"
              >
                <div
                  class="card"
                >
                  <img
                    alt="1+1"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="Джентльмены"
              >
                <div
                  class="card"
                >
                  <img
                    alt="Джентльмены"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Fd5cc31da-ec0c-4e4f-9282-614da6a5dbf2%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Fd5cc31da-ec0c-4e4f-9282-614da6a5dbf2%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Fd5cc31da-ec0c-4e4f-9282-614da6a5dbf2%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="Волк с Уолл-стрит"
              >
                <div
                  class="card"
                >
                  <img
                    alt="Волк с Уолл-стрит"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F40edbec8-9611-4a27-bd97-702c49f24df1%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F40edbec8-9611-4a27-bd97-702c49f24df1%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F40edbec8-9611-4a27-bd97-702c49f24df1%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="Гнев человеческий"
              >
                <div
                  class="card"
                >
                  <img
                    alt="Гнев человеческий"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F4303601%2F34b3860e-9074-43a3-8ce6-e9ca50ee4e6d%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F4303601%2F34b3860e-9074-43a3-8ce6-e9ca50ee4e6d%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F4303601%2F34b3860e-9074-43a3-8ce6-e9ca50ee4e6d%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="Брат"
              >
                <div
                  class="card"
                >
                  <img
                    alt="Брат"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2F471b833c-95b9-4095-866c-1d05c7769eb2%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2F471b833c-95b9-4095-866c-1d05c7769eb2%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2F471b833c-95b9-4095-866c-1d05c7769eb2%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="Один дома"
              >
                <div
                  class="card"
                >
                  <img
                    alt="Один дома"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F3f944968-628f-4030-9f74-5c77cd7d96c5%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F3f944968-628f-4030-9f74-5c77cd7d96c5%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F3f944968-628f-4030-9f74-5c77cd7d96c5%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="Аватар"
              >
                <div
                  class="card"
                >
                  <img
                    alt="Аватар"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F9c8a584d-05aa-4359-8aff-7d3c66671156%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F9c8a584d-05aa-4359-8aff-7d3c66671156%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F9c8a584d-05aa-4359-8aff-7d3c66671156%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="Побег из Шоушенка"
              >
                <div
                  class="card"
                >
                  <img
                    alt="Побег из Шоушенка"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fdfb73f87-b283-4150-9eba-fbbf3debaf04%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fdfb73f87-b283-4150-9eba-fbbf3debaf04%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fdfb73f87-b283-4150-9eba-fbbf3debaf04%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="Начало"
              >
                <div
                  class="card"
                >
                  <img
                    alt="Начало"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Ffde39e07-d0c4-4d24-8e66-7a5b9e184b85%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Ffde39e07-d0c4-4d24-8e66-7a5b9e184b85%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Ffde39e07-d0c4-4d24-8e66-7a5b9e184b85%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
            <div
              class="cardWrapper "
            >
              <a
                class="link"
                href="Зеленая миля"
              >
                <div
                  class="card"
                >
                  <img
                    alt="Зеленая миля"
                    class="img"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="1080"
                    src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fbc0fc6e3-fdbd-4f3e-bee4-9c6d3a1b5b62%2Forig&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fbc0fc6e3-fdbd-4f3e-bee4-9c6d3a1b5b62%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fbc0fc6e3-fdbd-4f3e-bee4-9c6d3a1b5b62%2Forig&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1920"
                  />
                </div>
              </a>
            </div>
          </div>
          <div
            class="carouselButtonWrapper"
          >
            <button
              class="carouselButton rightButton"
            >
              <img
                alt="arrow"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carouselContainer undefined"
    >
      <div
        class="titleContainer"
      />
      <div
        class="listContainer"
      >
        <div
          class="childrenContainer"
        >
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="1+1"
            >
              <div
                class="card"
              >
                <img
                  alt="1+1"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="Джентльмены"
            >
              <div
                class="card"
              >
                <img
                  alt="Джентльмены"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Fd5cc31da-ec0c-4e4f-9282-614da6a5dbf2%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Fd5cc31da-ec0c-4e4f-9282-614da6a5dbf2%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Fd5cc31da-ec0c-4e4f-9282-614da6a5dbf2%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="Волк с Уолл-стрит"
            >
              <div
                class="card"
              >
                <img
                  alt="Волк с Уолл-стрит"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F40edbec8-9611-4a27-bd97-702c49f24df1%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F40edbec8-9611-4a27-bd97-702c49f24df1%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F40edbec8-9611-4a27-bd97-702c49f24df1%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="Гнев человеческий"
            >
              <div
                class="card"
              >
                <img
                  alt="Гнев человеческий"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F4303601%2F34b3860e-9074-43a3-8ce6-e9ca50ee4e6d%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F4303601%2F34b3860e-9074-43a3-8ce6-e9ca50ee4e6d%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F4303601%2F34b3860e-9074-43a3-8ce6-e9ca50ee4e6d%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="Брат"
            >
              <div
                class="card"
              >
                <img
                  alt="Брат"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2F471b833c-95b9-4095-866c-1d05c7769eb2%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2F471b833c-95b9-4095-866c-1d05c7769eb2%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2F471b833c-95b9-4095-866c-1d05c7769eb2%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="Один дома"
            >
              <div
                class="card"
              >
                <img
                  alt="Один дома"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F3f944968-628f-4030-9f74-5c77cd7d96c5%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F3f944968-628f-4030-9f74-5c77cd7d96c5%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F3f944968-628f-4030-9f74-5c77cd7d96c5%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="Аватар"
            >
              <div
                class="card"
              >
                <img
                  alt="Аватар"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F9c8a584d-05aa-4359-8aff-7d3c66671156%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F9c8a584d-05aa-4359-8aff-7d3c66671156%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2F9c8a584d-05aa-4359-8aff-7d3c66671156%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="Побег из Шоушенка"
            >
              <div
                class="card"
              >
                <img
                  alt="Побег из Шоушенка"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fdfb73f87-b283-4150-9eba-fbbf3debaf04%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fdfb73f87-b283-4150-9eba-fbbf3debaf04%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fdfb73f87-b283-4150-9eba-fbbf3debaf04%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="Начало"
            >
              <div
                class="card"
              >
                <img
                  alt="Начало"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Ffde39e07-d0c4-4d24-8e66-7a5b9e184b85%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Ffde39e07-d0c4-4d24-8e66-7a5b9e184b85%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1773646%2Ffde39e07-d0c4-4d24-8e66-7a5b9e184b85%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
          <div
            class="cardWrapper "
          >
            <a
              class="link"
              href="Зеленая миля"
            >
              <div
                class="card"
              >
                <img
                  alt="Зеленая миля"
                  class="img"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="1080"
                  src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fbc0fc6e3-fdbd-4f3e-bee4-9c6d3a1b5b62%2Forig&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fbc0fc6e3-fdbd-4f3e-bee4-9c6d3a1b5b62%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2Fbc0fc6e3-fdbd-4f3e-bee4-9c6d3a1b5b62%2Forig&w=3840&q=75 2x"
                  style="color: transparent;"
                  width="1920"
                />
              </div>
            </a>
          </div>
        </div>
        <div
          class="carouselButtonWrapper"
        >
          <button
            class="carouselButton rightButton"
          >
            <img
              alt="arrow"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
